cherab/tools/observers/group/plotting.py

```diff
@@ -0,0 +1,157 @@
+# Copyright 2016-2021 Euratom
+# Copyright 2016-2021 United Kingdom Atomic Energy Authority
+# Copyright 2016-2021 Centro de Investigaciones Energéticas, Medioambientales y Tecnológicas
+#
+# Licensed under the EUPL, Version 1.1 or – as soon they will be approved by the
+# European Commission - subsequent versions of the EUPL (the "Licence");
+# You may not use this work except in compliance with the Licence.
+# You may obtain a copy of the Licence at:
+#
+# https://joinup.ec.europa.eu/software/page/eupl5
+#
+# Unless required by applicable law or agreed to in writing, software distributed
+# under the Licence is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR
+# CONDITIONS OF ANY KIND, either express or implied.
+#
+# See the Licence for the specific language governing permissions and limitations
+# under the Licence.
+
+import matplotlib.pyplot as plt
+from raysect.optical import Spectrum
+from raysect.optical.observer import RadiancePipeline0D, PowerPipeline0D, SpectralRadiancePipeline0D, SpectralPowerPipeline0D
+
+
+def select_pipelines(group, item):
+    """
+    Selects pipelines from the group based on index or name.
+
+    :param Observer0DGroup group: Observer group from which to select pipelines.
+    :param str/int item: The index or name of the pipeline to be selected.
+    :return list pipelines: matching pipelines
+    :return list observers: observers with matching pipelines
+    """
+    pipelines = []
+    observers = []
+    for observer in group._observers:
+        if isinstance(item, int):
+            if item < len(observer.pipelines):
```

Comment:
Raise an `IndexError` here: you can then explain exactly why no pipeline was found. You can either use this check or rely on the `IndexError` which would be returned when trying to access `observer.pipelines[item]`. Also, what happens if the index is negative? Is it what the user would expect? Especially if `abs(index) > len(observer.pipelines)`: you'll get an `IndexError` anyway.

Reply:
I agree with the comment on negative indexing, but I don't think that the `IndexError` should be raised. This function returns the lists of matching pipelines and the observers for which those pipelines are found. The observers in a group can have different number of pipelines, and if the _i_-th pipeline cannot be found for the _k_-th observer because this _k_-th observer has _N<sub>k</sub>_ < _i_ number of pipelines, then the _k_-th observer should be skipped because it has no matching pipeline.

Reply:
I have added a warning if invalid index is used.